homeassistant/components/sensor/foobot.py

```diff
@@ -0,0 +1,151 @@
+    dev = []
+    try:
+        devices = await client.get_devices()
+        _LOGGER.debug("The following devices were found: %s", devices)
+        for device in devices:
+            for sensor_type in SENSOR_TYPES:
+                if sensor_type == 'time':
+                    continue
+                foobot_sensor = FoobotSensor(client, data, device,
+                                             sensor_type)
+                dev.append(foobot_sensor)
+    except (aiohttp.client_exceptions.ClientConnectorError,
+            asyncio.TimeoutError, FoobotClient.TooManyRequests,
+            FoobotClient.InternalError):
+        _LOGGER.exception('Failed to connect to foobot servers.')
+        raise PlatformNotReady
+    except FoobotClient.ClientError:
+        _LOGGER.error('Failed to fetch data from foobot servers.')
+        return
+    async_add_devices(dev, True)
+
+
+class FoobotSensor(Entity):
+    """Implementation of a Foobot sensor."""
+
+    __cache = {}
+    __nb_requests = 0
+
+    def __init__(self, client, data, device, sensor_type):
+        """Initialize the sensor."""
+        self._client = client
+        self._uuid = device['uuid']
+        self._name = 'Foobot {} {}'.format(device['name'],
+                                           SENSOR_TYPES[sensor_type][0])
+        self.type = sensor_type
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return SENSOR_TYPES[self.type][2]
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        try:
+            data = self.__cache[self._uuid][self.type]
+        except(KeyError, TypeError):
+            data = None
+        return data
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    async def async_update(self):
+        """Get the latest data."""
+        if self._uuid in self.__cache:
+            last_request = self.__cache[self._uuid]['last_request']
+            if last_request + SCAN_INTERVAL > datetime.utcnow():
+                # If the last request made is recent enough,
+                # just skip the update
+                _LOGGER.debug('Skipping update')
+                return
+
+        interval = SCAN_INTERVAL.total_seconds()
+        try:
+            response = await self._client.get_last_data(self._uuid,
+                                                        interval,
+                                                        interval + 1)
+        except (aiohttp.client_exceptions.ClientConnectorError,
+                asyncio.TimeoutError, self._client.TooManyRequests,
+                self._client.InternalError):
+            _LOGGER.debug("Couldn't fetch data")
+        if response:
```

Comment:
`response` will never have been assigned  if the exception happened. Just add a return inside the except block and drop the if.